### DISCUSSION TITLE
Mapping tuples and anonymous records

### DISCUSSION THREAD

**Discussion by Xyncgas** [fsharp/fslang-suggestions#1295] [Library]
I request the F# language supports converting a tuple of Async items to Async tuple of items, no matter how many items are in the tuple

Maybe consider adding support to map (typeA<'a>,typeA<'b>,typeA<'c>,typeA<'d>)... to ('a,'b'c,'d)...
or ('a,'b,'c,'d) to ('a,'b,'c,'d) via Tuple.Map which is something we don't have in the library

**Comment by smoothdeveloper** [comment]
Wondering if higher kinded type enable something generic to traverse structure generically and do such transformation?

**Comment by Xyncgas** [comment]
Updated title because it's similar, if we can write tuple / anonymous record (and struct equivalent) and use a function map on it, it would be lesser we write while seamlessly working with functional style
